aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,312 @@
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/AddHeaderWithFactoryAttribute.cs
+  :language: c#
+  :emphasize-lines: 1,3-6
+  :lines: 6-25
+  :dedent: 4
+
+@rynowak - can you clarify this next bit?
+If you're using ``IFilterFactory``, you can specify lifetime of the filter (how?). When your filter is the instance, your filter instance is cached, so avoid do anything stateful in the filter returned by ``CreateInstance``.
+ 
+Ordering
+^^^^^^^^
+Filters can be applied per-action method (via attribute) or via controller (via attribute), or in global filters collection. Scope also generally determines ordering. The filter closest to the action runs first; generally you get overriding behavior without having to explicitly set ordering. This is sometimes referred to as "russian doll" nesting, as each increase in scope is wrapped around the previous scope, like a `nesting doll <https://en.wikipedia.org/wiki/Matryoshka_doll>`_.
+
+In addition to scope, filters can override their sequence of execution by implementing `IOrderedFilter <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Filters/IOrderedFilter/index.html>`_. This interface simply exposes an ``int`` ``Order`` property, and filters execute in ascending numeric order based on this property. All of the built-in filters, including ``TypeFilterAttribute`` and ``ServiceFilterAttribute``, implement ``IOrderedFilter``, so you can specify the order of filters when you apply the attribute to a class or method.
+
+To see the scope-based ordering in action, consider the following class, which has the same filter applied at both the class and method level. The same filter is also registered globally. When run, the log output will show the order in which the filters executed, and how these compared to the ``Controller`` base class's ``OnActionExecuting`` and ``OnActionExecuted`` methods.
+
+.. literalinclude:: filters/sample/src/FiltersSample/Controllers/OrderingController.cs
+  :language: c#
+  :emphasize-lines: 1,10,18,24
+  :lines: 8-34
+  :dedent: 4
+
+The resulting order is:
+	1. The Controller ``OnActionExecuting``
+	2. The Global filter ``OnActionExecuting``
+	3. The Class filter ``OnActionExecuting``
+	4. The Method filter ``OnActionExecuting``
+	5. The Method filter ``OnActionExecuted``
+	6. The Class filter ``OnActionExecuted``
+	7. The Global filter ``OnActionExecuted``
+	8. The Controller ``OnActionExecuted``
+
+.. note:: ``IOrderedFilter`` trumps scope when determining the order in which filters will run. Filters are sorted first by order, then scope is used to break ties. Order defaults to 0 if not set.
+
+To modify the default, scope-based order, you could explicitly set the ``Order`` property of the class-level and/or method-level filters. For example, ``[OrderLoggingActionFilter(Name = "Method Level Attribute", Order=-1)]``. In this case, a value of less than zero would ensure this filter ran before both the Global and Class level filters.
```

Comment:
>For example, ``[OrderLoggingActionFilter(Name = "Method Level Attribute", Order=-1)]``. I

For example, adding  Order=-1 to the action method:

``[OrderLoggingActionFilter(Name = "Method Level Attribute", Order=-1)]``